compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -126,6 +134,8 @@ struct LoweringContext<'a, 'hir> {
     impl_trait_defs: Vec<hir::GenericParam<'hir>>,
     impl_trait_bounds: Vec<hir::WherePredicate<'hir>>,
 
+    in_scope_generics: Vec<LocalDefId>,
```

Comment:
Why do we need this?

Reply:
This is used to keep track of in scope lifetimes and then filter the ones that we've already created defs for (the ones that show up in hir bounds) to lower those lifetimes as `static` https://github.com/rust-lang/rust/pull/101345/files#diff-ad0c15bbde97a607d4758ec7eaf88248be5d6b8ae084dfc84127f81e3f7a9bb4R2472. I guess you already know about this thing that is being done, this thing exist in master and we would need at some point to remove this hack.